File: 0496-next-greater-element-i/0496-next-greater-element-i.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        vector<int> greater_el;
        for(int i = 0; i < nums1.size(); i++){
            int mx = nums1[i];
            bool met = false;
            for(int j = 0; j < nums2.size(); j++){
                if(met && nums2[j] > mx){
                    mx = nums2[j];
                    break;
                }
                if(nums1[i] == nums2[j]){
                    met = true;
                }
            }
            greater_el.push_back(mx == nums1[i] ? -1 : mx);
        }
        return greater_el;
    }
};

File: 0496-next-greater-element-i/0496-next-greater-element-i.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        stack = [nums2[0]]
        mp = {}
        for i in range(1, len(nums2)):
            while stack and stack[-1] < nums2[i]:
                mp[stack.pop()] = nums2[i]
            stack.append(nums2[i])
        
        for st in stack:
            mp[st] = -1
        ans = []
        for i in nums1:
            ans.append(mp[i])
        return ans